Sync with latest addon blueprint and use decorator-transforms

SergeAstapov opened the pull request:
v2 addon blueprint: https://github.com/embroider-build/addon-blueprint

Minor gain: using `decorator-transforms` reduced bundle size for the app.

SergeAstapov commented:
side note: funnily enough, v5 scenarios pass only because repo uses yarn and yarn incorrectly handles peer dependencies

villander commented on .github/workflows/ci.yml:
```diff
@@ -73,25 +67,24 @@ jobs:
-          - ember-classic
-#          - embroider-safe
-#          - embroider-optimized
+          # - embroider-safe
+          # - embroider-optimized
```
IMO does not make sense sync with embroider blueprint and don't support it on CI

SergeAstapov replied:
@villander as this was commented out, I didn't touch. Let me open separate PR and uncomment these scenarios. I'm not sure why they been commented and hat was there error

villander replied:
Oh sorry! nevermind

villander replied:
@SergeAstapov have you tried enable them to make it compatible with embroider?

SergeAstapov replied:
@villander no, it's out of scope of this PR.

villander commented on README.md:
```diff
@@ -7,26 +7,20 @@ This addon provides an API for authoring a [Router service](https://api.emberjs.
-* Node.js v14 or above
 
+- Ember.js v3.12 or above
+- Embroider or ember-auto-import v2
```
Could you please link the embroider blueprint to the PR description for better testing and review @SergeAstapov ?

SergeAstapov replied:
@villander this was already done

villander reviewed (changes requested):
Couple requested changes! 

@SergeAstapov could you please describe what's size do you mean about the app? Are you talking about the testing-app?

SergeAstapov commented:
> Couple requested changes!
> 
> @SergeAstapov could you please describe what's size do you mean about the app? Are you talking about the testing-app?

added link to PR description.

RE: `decorator-transforms/`
You may read about this package in https://github.com/ef4/decorator-transforms/
now it comes from v2 addon blueprint, see https://github.com/embroider-build/addon-blueprint/blob/main/files/__addonLocation__/package.json#L37

TL;DR `@babel/plugin-proposal-decorators` has large footprint in terms of compiled code size. In our app we saw 18% reduction in size for `vendor.js` and `app.js` once we introduced `decorator-transforms/` and disabled `@babel/plugin-proposal-decorators`.

villander commented on test-app/ember-cli-build.js:
```diff
@@ -4,20 +4,11 @@ const EmberApp = require('ember-cli/lib/broccoli/ember-app');
   const app = new EmberApp(defaults, {
-    // Add options here
+    autoImport: {
+      watchDependencies: ['ember-engines-router-service'],
```
@SergeAstapov why do we need this config?

SergeAstapov replied:
@villander this tells `ember-auto-import` to pick up changes in the addon.

This makes `ember s` and `ember test --server` commands pick up changes in the `ember-engines-router-service` addon without restarting the command.

This is not unique to `ember-engines-router-service`, it's just a config now aded by blueprint https://github.com/embroider-build/addon-blueprint/blob/main/files/test-app-overrides/ember-cli-build.js#L9 and that was just not present in this repo before.

villander replied:
@SergeAstapov could you please enable it only for non-production environments?

SergeAstapov replied:
@villander this flag only affects `ember test --server` and `ember server` commands, hence this makes no effect for `ember build` command.

villander replied:
Thank you @SergeAstapov

villander commented:
We can merge this PR when CI get green again

SergeAstapov commented:
@villander this is refreshed with the latest blueprint as of now and CI is passing

villander reviewed (approved):
Thank you! Sorry for the delay in merging it
